src/librustc_error_codes/error_codes/E0657.md

```diff
@@ -0,0 +1,64 @@
+The `impl Trait` can only capture lifetimes bound at the fn or impl level.
+
+Erroneous code example:
+
+```compile_fail,E0657
+#![allow(warnings)]
+
+trait Id<T> {}
```

Comment:
You need to add a comment (at minimum `// error!`) on the failing line to help users identify the problem more visually.